handler/src/main/java/io/netty/handler/stream/ChunkedWriteHandler.java

```diff
@@ -125,7 +125,10 @@ private void resumeTransfer0(ChannelHandlerContext ctx) {
 
     @Override
     public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
-        queue.add(new PendingWrite(msg, promise));
+        if (queue == null) {
+            queue = new PendingWriteQueue(ctx);
```

Comment:
imho to simplify this we should just create the queue in `handlerAdded(...)`

Reply:
I will but I'm not yet fully convinced about this change: 

- the existing version could lead to OOM or chunk send starvation with both a huge number of pending `writeAndFlush` events with `ByteBuf` msgs mixed with `ChunkedInput` ones
- this PR version could lead to OOM due to slow network if `ChunkedInput` pending flushes got too high

I would like to save the best from both